### PR TITLE
Add truffle library documentation

### PR DESCRIPTION
**Pull request by cgewecke**
Documentation for [truffle 2226](https://github.com/trufflesuite/truffle/pull/2226)

cc: @gnidan

### CODE REVIEW

File: src/docs/truffle/getting-started/writing-external-scripts.md [hunk 1]
```diff
@@ -80,7 +80,7 @@ know how we can improve it!
      // config_[1] starts remaining parameters.
      if (config.help) {
        console.log(`Usage: truffle run hello [name]`);
-       done(null, [], []);	
+       done(null, [], []);
```
Comment (cgewecke) [inline src/docs/truffle/getting-started/writing-external-scripts.md]: My editor made this spacing change, just noticed it. Is this line a typo anyway though?

Comment (eggplantzzz) [reply src/docs/truffle/getting-started/writing-external-scripts.md]: This line looks like a typo to me, `done` isn't part of the method signature. @gnidan @CruzMolina

Comment (cgewecke) [reply src/docs/truffle/getting-started/writing-external-scripts.md]: @eggplantzzz Ok .. removing.

File: src/docs/truffle/getting-started/writing-external-scripts.md [hunk 2]
```diff
@@ -105,3 +105,41 @@ know how we can improve it!
...
+   measure and save gas usage data
+
+aggregate data and report
+```
+The Truffle library lets you do this without making the user add configuration or string their own commands together.
+
+:warning: **Important Note** :warning:
+
+Truffle does not guarantee its internal APIs will follow semver. You should be prepared for your user
+to run any Truffle version and handle mismatches gracefully. By using the library **you are entering into a gentleperson's agreement** to manage API volatility and other contingencies on your user's behalf. Some tips:
```
[the remark this answers is not in the document]
Comment (cgewecke) [reply src/docs/truffle/getting-started/writing-external-scripts.md]: @gnidan How do you do "suggested change"?  Is that just an option in the review comment? (It's super cool)

**Pull request merged**